src/core/tools/types.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * Context passed to tool execution
+ */
+export interface ToolExecutionContext {
+    /** Session ID if available */
+    sessionId?: string | undefined;
+}
+
+/**
+ * Simple tool interface for internal tools
+ */
+export interface Tool {
+    /** Unique identifier for the tool */
+    id: string;
+
+    /** Human-readable description of what the tool does */
+    description: string;
+
+    /** Zod schema defining the input parameters */
+    inputSchema: any; // Using any for flexibility since we don't import zod here
+
+    /** The actual function that executes the tool */
+    execute: (input: any, context?: ToolExecutionContext) => Promise<any> | any;
+}
+
+/**
+ * Tool parameters interface for normalized tool definitions
+ */
+export interface ToolParameters {
+    type: 'object';
+    properties: Record<string, unknown>;
+    required?: string[];
+    [key: string]: unknown; // Allow additional JSON Schema properties
+}
+
+/**
+ * Raw tool definition interface (before normalization)
+ */
+export interface RawToolDefinition {
+    description?: string;
+    parameters?: {
+        type?: string;
+        properties?: Record<string, unknown>;
+        required?: string[];
+    };
+}
+
+/**
+ * Tool manager's unified tool set (for compatibility with MCP tools and ai/types.ts)
+ */
+export interface ToolManagerToolSet {
+    [toolName: string]: {
+        name?: string;
+        description: string;
+        parameters?: ToolParameters;
+    };
+}
```

Comment:
we already have most of these types in this file in core/ai/types.ts

we can move those types into here and add the new ones, these are mostly duplicate

there are also some small subtle changes in these duplicate types that are leading to a whole lot of unnecessary code - `ToolManagerToolSet` is the same as `ToolSet` except that description is no longer optional.

this is resulting in `ToolManager.getMcpTools()`, which should be just one line calling `McpManager.getAllTools()` needing unnecessary type conversion between the two ToolSet types. just using ToolSet everywhere would have avoided this

there are some other problems, can we make a gh issue for this, i will fix in follow up